From stable-bounces@linux.kernel.org  Sun Jun 25 03:05:24 2006
Date: Sun, 25 Jun 2006 12:04:36 +0200
From: Sam Ravnborg <sam@ravnborg.org>
To: stable@kernel.org
Cc: 
Subject: kbuild: Fix 100% initramfs bloat in 2.6.17 versus 2.6.16

From: Nix <nix@esperi.org.uk>

When I built 2.6.17 for the first time I was a little surprised to see
my kernel putting on >500Kb in weight.

It didn't take long to work out that this was because my initramfs's
contents were being included twice in the cpio image.

A make V=1 makes the problem obvious:

/bin/sh /usr/packages/linux/versions/i686-loki/scripts/gen_initramfs_list.sh -l  "usr/initramfs" > usr/.initramfs_data.cpio.gz.d
  /bin/sh /usr/packages/linux/versions/i686-loki/scripts/gen_initramfs_list.sh -o usr/initramfs_data.cpio.gz  -u 0  -g 0  "usr/initramfs"  "usr/initramfs"

Note that doubling-up of the "usr/initramfs", which leads to
gen_initramfs_list.sh dumping the thing into the cpio archive twice.

The cause is an obvious pasto, fixed thusly:

Signed-off-by: Nick Alcock <nix@esperi.org.uk>
Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---
 usr/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.17.1.orig/usr/Makefile
+++ linux-2.6.17.1/usr/Makefile
@@ -33,7 +33,7 @@ ifneq ($(wildcard $(obj)/.initramfs_data
 endif
 
 quiet_cmd_initfs = GEN     $@
-      cmd_initfs = $(initramfs) -o $@ $(ramfs-args) $(ramfs-input)
+      cmd_initfs = $(initramfs) -o $@ $(ramfs-args)
 
 targets := initramfs_data.cpio.gz
 $(deps_initramfs): klibcdirs
